ipc/cli/src/commands/subnet/join.rs

```diff
@@ -136,3 +147,45 @@ pub struct UnstakeSubnetArgs {
     )]
     pub collateral: f64,
 }
+
+pub struct PreFundSubnet;
+
+#[async_trait]
+impl CommandLineHandler for PreFundSubnet {
+    type Arguments = PreFundSubnetArgs;
+
+    async fn handle(global: &GlobalArguments, arguments: &Self::Arguments) -> anyhow::Result<()> {
+        log::debug!("pre-fund subnet with args: {:?}", arguments);
+
+        let mut provider = get_ipc_provider(global)?;
+        let subnet = SubnetID::from_str(&arguments.subnet)?;
+        let from = match &arguments.from {
+            Some(address) => Some(require_fil_addr_from_str(address)?),
+            None => None,
+        };
+        provider
+            .prefund_subnet(
+                subnet.clone(),
+                from,
+                f64_to_token_amount(arguments.initial_balance)?,
+            )
+            .await?;
+        println!("address pre-funded successfully");
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Args)]
+#[command(
+    name = "pre-fund",
+    about = "Pre fund with some funds in genesis in a child-subnet"
+)]
+pub struct PreFundSubnetArgs {
+    #[arg(long, short, help = "The address funded in the subnet")]
+    pub from: Option<String>,
+    #[arg(long, short, help = "The subnet to add balance to")]
+    pub subnet: String,
+    #[arg(help = "Optionally add an initial balance to the validator in genesis in the subnet")]
```

Comment:
It's not optional here. Can only validators use this?

Reply:
Typo due to copy-past. Fixed! Thanks!